### ISSUE TITLE
`nt.empirical_ntk_fn` and `nt.linearize` do not compute gradients past the linear head in a ViT

### ISSUE THREAD

**Issue by refitwhisper** [google/neural-tangents#149]
Hi, thanks for your excellent codebase. I am trying to train a (finite-width) linearized vision transformer (which I have obtained from the [vision_transformer](https://github.com/google-research/vision_transformer) repo) using your `nt.linearize` function. I am closely following the example in the [Weight Space Linearization colab](https://colab.research.google.com/github/google/neural-tangents/blob/main/notebooks/weight_space_linearization.ipynb), but I have noticed that the gradients I obtain are all equal to zero, except those in the linear readout ("head") layer. So training the "linearized" model just amounts to training the linear head.

Similarly, I find that the empirical kernel computed using `nt.empirical_ntk_fn` does not make use of the full Jacobian of the network, but instead only the Jacobian of the linear head. So the empirical kernel I obtain is actually just identical to the gram matrix of the activations in the embedding layer.

How can I make sure that `nt.linearize` and `nt.empirical_ntk_fn` are computing gradients for all parameters in the ViT, and not just those in the linear head?

Here is a [minimal colab](https://colab.research.google.com/drive/184moQLq3tjo-wEpc8gD7fXCFguAVDBOm?usp=sharing) that reproduces the issue (this colab simply combines the [Weight Space Linearization colab](https://colab.research.google.com/github/google/neural-tangents/blob/main/notebooks/weight_space_linearization.ipynb) from this repo with the  [Vision Transformer](https://colab.research.google.com/github/google-research/vision_transformer/blob/master/vit_jax.ipynb) colab from the vision_transformer repo).

Thank you!

**Comment by Sohl-Dickstein** [comment]
It looks like the readout weights of that architecture are initialized to 0:
https://github.com/google-research/vision_transformer/blob/67e9aba949e55948974707860f787330c893c0a5/vit_jax/models.py#L296

If the readout weights are zero, then the gradient with respect to all the
earlier parameters in the network will also be 0 (you can see this by
noting that if you use backprop to compute the parameter gradients, you
will multiply by the transpose of the (zero) readout weights). This is why
you only get a nonzero gradient with respect to the readout weights.

If you would like nonzero gradients at init with respect to earlier layers,
you should choose a different readout weight initialization.

Hope this is helpful!

**Comment by refitwhisper** [comment]
ah great catch, thank you!
